### DISCUSSION TITLE
Where the omnisharp log in VSCode?

### DISCUSSION THREAD

**Discussion by ProgrammingLife** [OmniSharp/omnisharp-vscode#7604] [General]
I opened my project from Unity3D it installed all plugins successfully. Looks like omnisharp was installed too.
Autocompletion works fine. I see all the Unity3D hints for functions and methods.
```bash
$ find ~/.vscode/extensions/ -iname 'ms-*' 
/home/user/.vscode/extensions/ms-dotnettools.vscode-dotnet-runtime-2.1.6
/home/user/.vscode/extensions/ms-dotnettools.csdevkit-1.10.18-linux-x64
/home/user/.vscode/extensions/ms-dotnettools.csharp-2.45.25-linux-x64
```

When I look at dropdown list placed in Output tab, there's no anything that associated with Omnisharp name. That's the first problem. And the second one is ignoring omnisharp formatting settings by Omnisharp which were placed here:
```bash
~/.omnisharp/omnisharp.json
<project>/omnisharp.json
<project>/.editorconfig
```
All files with the same contents. No one of those file worked.

Tried to change `"omnisharp.enableEditorConfigSupport"` - nothing works. And I even don't see the logs again.
In my vscode settings.json I have settings:
```
	"omnisharp.loggingLevel": "debug",

	"razor.format.enable": true,
	"csharp.format.enable": true,
	"omnisharp.useModernNet": false,
	"omnisharp.monoPath": "/usr/bin/mono",
	"omnisharp.enableEditorConfigSupport": false,
	"omnisharp.useEditorFormattingSettings": true
```

Even if I press: ctrl+shift+p looking for `restart omnisharp` - I don't even see anything associated with omnisharp text.

All my extenions:
```
$ ls -1 ~/.vscode/extensions/ 
airyshift.spaces-inside-braces-0.4.0/
extensions.json
ms-dotnettools.csdevkit-1.10.18-linux-x64/
ms-dotnettools.csharp-2.45.25-linux-x64/
ms-dotnettools.vscode-dotnet-runtime-2.1.6/
rokoroku.vscode-theme-darcula-1.2.3/
visualstudiotoolsforunity.vstuc-1.0.4/

$ dotnet --version
8.0.108

$ mono --version
Mono JIT compiler version 6.12.0 (makepkg/0cbf0e290c3 Sat Mar  9 11:37:33 UTC 2024)
```

In the Output tab -> "C#" I see such logs:
```
Using dotnet configured on PATH
Dotnet path: /usr/share/dotnet/dotnet
Activating C# + C# Dev Kit...
waiting for named pipe information from server...
[stdout] {"pipeName":"/tmp/854d4bc4.sock"}
received named pipe information from server
attempting to connect client to server...
client has connected to server
[Info  - 1:58:27 PM] [Program] Language server initialized
[Info  - 1:58:31 PM] [WorkspaceProjectFactoryService] Project /home/user/projects/unity/MyFirstTry_3d/Assembly-CSharp.csproj loaded by C# Dev Kit
```

What steps should I check to find out what went wrong?

Arch Linux, VSCode 1.93.1, Unity3D 6 Preview.

**Comment by dibarbet** [comment]
Apologies for the delay here.  In your case it looks like you're not using OmniSharp, you're using the new version of the extension powered by Roslyn.  It has a different set of log files and does not use omnisharp.json.   The Roslyn version does support editorconfig and loading unity projects.  If you're having an issue with it, please file an issue.

If you need O# for a specific reason, you can tell the extension to use O# instead by setting `dotnet.server.useOmnisharp` to true and disabling the C# Dev Kit extension.
